airflow/lineage/__init__.py

```diff
@@ -142,7 +142,7 @@ def wrapper(self, context, *args, **kwargs):
                 _inlets = self.xcom_pull(
                     context, task_ids=task_ids, dag_id=self.dag_id, key=PIPELINE_OUTLETS, session=session
                 )
-                self.inlets.extend(i for i in itertools.chain.from_iterable(_inlets))
+                self.inlets.extend(itertools.chain.from_iterable(_inlets))
```

Comment:
This can just be

```suggestion
                self.inlets.extend(i for it in _inlets for i in it)
```

but itertools has readability advantages.